compiler/rustc_span/src/lib.rs

```diff
@@ -1745,6 +1745,21 @@ impl SourceFile {
         BytePos::from_u32(pos.0 - self.start_pos.0 + diff)
     }
 
+    /// Calculates a normalized byte position from a byte offset relative to the
+    /// start of the file.
+    pub fn normalized_byte_pos(&self, offset: u32) -> BytePos {
```

Comment:
Not really familiar with this code. I don't know what normalization in this context is, so I don't really understand what this function does. I also find it kind of surprising that functionality like this hasn't existed before and you need to implement it yourself to adjust some spans.

Feel free to assign to somebody else, if you don't have the time to explain what normalization is in this context (I don't have the time to read up on it unfortunately).

Reply:
Normalization refers to [this function](https://github.com/Amanieu/rust/blob/07b33b611aba4806087019d868241b6ac798eb8a/compiler/rustc_span/src/lib.rs#L1847) which removes the unicode BOM and converts `\n\r` newlines to `\r` when a new `SourceFile` is added.

When we get an inline assembler error from LLVM during codegen, we import the expanded assembly code as a new `SourceFile`, which can then be used for error reporting with spans. However the byte offsets given to us by LLVM are relative to the start of the original buffer, not the normalized one. Hence we need to convert those offsets to the normalized form when constructing spans.

Reply:
Thanks, makes sense to me now. Can you maybe add the second part of your comment to the docs for that function?

Reply:
I've added the comment.